src/render/gpu/SDL_render_gpu.c

```diff
@@ -1228,19 +1326,26 @@ static bool GPU_CreateRenderer(SDL_Renderer *renderer, SDL_Window *window, SDL_P
         return false;
     }
 
-    if (!SDL_ClaimWindowForGPUDevice(data->device, window)) {
+    if (data->user_managed_device && !InitPresentQuadBuffer(data)) {
         return false;
     }
 
+    if (!data->user_managed_device) {
+        if (!SDL_ClaimWindowForGPUDevice(data->device, window)) {
+            return false;
+        }
+    }
+
     data->swapchain.composition = SDL_GPU_SWAPCHAINCOMPOSITION_SDR;
     data->swapchain.present_mode = SDL_GPU_PRESENTMODE_VSYNC;
 
-    int vsync = (int)SDL_GetNumberProperty(create_props, SDL_PROP_RENDERER_CREATE_PRESENT_VSYNC_NUMBER, 0);
-    ChoosePresentMode(data->device, window, vsync, &data->swapchain.present_mode);
-
-    SDL_SetGPUSwapchainParameters(data->device, window, data->swapchain.composition, data->swapchain.present_mode);
+    if (!data->user_managed_device) {
```

Comment:
I am not sure if we should play with vsync settings or present mode in case the device is not ours